oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/client/RegisteredClientTests.java

```diff
@@ -309,7 +309,7 @@ public void buildWhenClientAuthenticationMethodsConsumerIsProvidedThenConsumerAc
 				.build();
 
 		assertThat(registration.getClientAuthenticationMethods())
-				.containsExactlyInAnyOrder(ClientAuthenticationMethod.BASIC, ClientAuthenticationMethod.POST);
+				.containsExactlyInAnyOrder(ClientAuthenticationMethod.CLIENT_SECRET_BASIC, ClientAuthenticationMethod.CLIENT_SECRET_POST);
```

Comment:
I think we should add an additional test here that uses the non-deprecated methods, but retain this one to test the upgrade process. Perhaps this one could be renamed to indicate it's testing the upgrade. `buildWhenConsumerAddsDeprecatedClientAuthenticationMethodsThenNonDeprecatedAreRegistered`.

Reply:
I have added the extra test `buildWhenConsumerAddsDeprecatedClientAuthenticationMethodsThenNonDeprecatedAreRegistered `